src/circuits/joinsplit.tcc

```diff
@@ -118,60 +126,59 @@ public:
+            //   + 1,  NumOutputs + NumInputs + 1 + NumInputs[
+            // - Index of the "v_pub_in", "v_pub_out", and bits of previous
+            //   varibles not fitting within FieldT::capacity() [NumOutputs +
+            //   NumInputs + 1 + NumInputs, ?]
+            //
             // We first allocate the root
             merkle_root.reset(new libsnark::pb_variable<FieldT>);
             merkle_root->allocate(
                 pb, FMT(this->annotation_prefix, " merkle_root"));
 
-            // We allocate 2 field elements to pack each inputs nullifiers and
+            // We allocate a field element to pack each inputs nullifiers and
             // each output commitments
             for (size_t i = 0; i < NumInputs; i++) {
                 packed_inputs[i].allocate(
                     pb,
-                    1 + 1,
+                    1,
                     FMT(this->annotation_prefix, " in_nullifier[%zu]", i));
             }
             for (size_t i = NumInputs; i < NumInputs + NumOutputs; i++) {
                 packed_inputs[i].allocate(
                     pb,
-                    1 + 1,
+                    1,
                     FMT(this->annotation_prefix, " out_commitment[%zu]", i));
             }
 
-            // We allocate 1 field element to pack the value (v_pub_in)
+            // We allocate a field element to pack the value of h_sig
             packed_inputs[NumInputs + NumOutputs].allocate(
-                pb, 1, FMT(this->annotation_prefix, " v_pub_in"));
-
-            // We allocate 1 field element to pack the value (v_pub_out)
-            packed_inputs[NumInputs + NumOutputs + 1].allocate(
-                pb, 1, FMT(this->annotation_prefix, " v_pub_out"));
+                pb, 1, FMT(this->annotation_prefix, " h_sig"));
 
-            // We allocate 2 field elements to pack the value of h_sig
-            packed_inputs[NumInputs + NumOutputs + 1 + 1].allocate(
-                pb, 1 + 1, FMT(this->annotation_prefix, " h_sig"));
-
-            // We allocate 2 field elements to pack each malleability tags h_iS
-            for (size_t i = NumInputs + NumOutputs + 1 + 1 + 1;
-                 i < NumInputs + NumOutputs + 1 + 1 + 1 + NumInputs;
+            // We allocate a field element to pack each malleability tags h_iS
+            for (size_t i = NumInputs + NumOutputs + 1;
+                 i < NumInputs + NumOutputs + 1 + NumInputs;
                  i++) {
                 packed_inputs[i].allocate(
-                    pb, 1 + 1, FMT(this->annotation_prefix, " h_i[%zu]", i));
+                    pb, 1, FMT(this->annotation_prefix, " h_i[%zu]", i));
             }
 
+            // We allocate as many field elements as needed to pack the public
+            // values and unpacked residual bits
+            packed_inputs[NumInputs + NumOutputs + 1 + NumInputs].allocate(
+                pb,
+                (residual_packing_bits + log_r - 1) / log_r,
```

Comment:
Why not use `libff::div_ceil` and the field capacity `FieldT::capacity`?
ie: `libff::div_ceil(residual_packing_bits, FieldT::capacity());`

Reply:
Done